lookout/style/format/feature_extractor.py

```diff
@@ -358,6 +344,84 @@ def composite_class_printables(self) -> List[str]:
         return ["".join(CLASS_PRINTABLES[label] for label in labels)
                 for labels in self.labels_to_class_sequences]
 
+    def _create_neighbours(self, vnodes: Sequence[VirtualNode], vnodes_y: Sequence[VirtualNode],
+                           parents: Mapping[int, bblfsh.Node], return_sibling_indices: bool = False
+                           ) -> Tuple[Layout[Sequence[Optional[AnyNode]]],
+                                      Optional[List[List[int]]]]:
+        if self.return_sibling_indices:
+            sibling_indices_list = []
+        neighbours = OrderedDict()  # type: MutableLayout[List[Optional[AnyNode]]]
+        for group in FeatureGroup:
+            neighbours[group] = []
+        neighbours[FeatureGroup.node].append([])
+        for _ in range(self.left_siblings_window):
+            neighbours[FeatureGroup.left].append([])
+        for _ in range(self.right_siblings_window):
+            neighbours[FeatureGroup.right].append([])
+        for _ in range(self.parents_depth):
+            neighbours[FeatureGroup.parents].append([])
+
+        vnodes_y_set = frozenset(id(vnode_y) for vnode_y in vnodes_y)
+        closest_left_node_id = None
+
+        for i, vnode in enumerate(vnodes):
+            if vnode.node:
+                closest_left_node_id = id(vnode.node)
+            if id(vnode) not in vnodes_y_set:
+                continue
```

Comment:
it is kind of computation heavy.
we know that `vnode` and `vnode_y` have the same order, so we can compare with only one element from `vnode_y`
```python
 if vnode != vnode_y[vnode_y_index]:
    vnode_y_index += 1
    continue
```

Reply:
It's not really computation heavy though. Set creation is linear on the size of vnode_y and in operation is O(1) so both your version and this one are overall linear on the size of vnode_y. I think therefore we should decide mainly on readability and maintainability and I feel like the PR version is clearer and doesn't maintain an additional counter that can be moved/messed with that could introduce bug.